CMakeLists.txt

```diff
@@ -113,12 +123,14 @@ if(CMAKE_C_COMPILER_ID MATCHES "GNU|AppleClang|Clang")
   #
   # -Wconversions
 
-  add_compile_options(-Wempty-body -Wvla)
-  check_c_compiler_flag(-Wimplicit-fallthrough CC_SUPPORTS_IMPLICIT_FALLTHROUGH)
+  # These flags are supported on all compilers.
+  add_compile_options(
```

Comment:
Will it be better to use `add_compile_options_if_supported` here? Or is it more important that these flags are supported and fail otherwise?

Reply:
We can probably use `add_compile_options_if_supported`, but that is a more expensive operation, so if we have too many flags it will be slow.

Reply:
No strong opinion, but using `add_compile_options_if_supported` works here as well, but it is a (relatively) time-consuming operation since it will test the flag with the compiler. If we have flags that are known to be supported on all, it's better to not using this function.

Reply:
Then I guess it can be kept as it is.